src/api/spec/controllers/webui/request_controller/show_spec.rb

```diff
@@ -0,0 +1,415 @@
+require 'rails_helper'
+
+RSpec.describe Webui::RequestController, '#show', vcr: true do
+  let(:submitter) { create(:confirmed_user, :with_home, login: 'kugelblitz') }
+  let(:receiver) { create(:confirmed_user, :with_home, login: 'titan') }
+  let(:target_project) { receiver.home_project }
+  let(:target_package) { create(:package_with_file, name: 'goal', project_id: target_project.id) }
+  let(:source_project) { submitter.home_project }
+  let(:source_package) { create(:package, :as_submission_source, name: 'ball', project: source_project) }
+  let(:bs_request) do
+    create(:bs_request_with_submit_action,
+           description: 'Please take this',
+           creator: submitter,
+           target_package: target_package,
+           source_package: source_package)
+  end
+
+  describe 'GET #show' do
+    context 'when the user is part of the beta program' do
+      render_views
+
+      before do
+        Flipper.enable(:request_show_redesign, receiver)
+      end
+
+      context 'when the user has only one submit action' do
+        let!(:comment) { create(:comment, commentable: bs_request) }
+
+        before do
+          login receiver
+          get :show, params: { number: bs_request.number }
+        end
+
+        it 'shows the new redesign page' do
+          expect(controller).to render_template('webui/request/beta_show')
+          expect(response.body).to have_text("Request #{bs_request.number}")
+        end
+
+        it 'shows the history elements for the request' do
+          expect(response.body).to have_text("Created by\n #{submitter.name}")
+          expect(response.body).to have_text("(#{bs_request.creator})\ncreated this request")
+        end
+
+        it 'shows the request comment' do
+          expect(response.body).to have_selector('.timeline-item', text: comment.body)
+        end
+
+        it { expect(response.body).to have_text("The RPM Lint results aren't here yet") }
+        it { expect(response.body).to have_text("The mentioned issues aren't here yet") }
+      end
+
+      context 'when the request has multiple submit actions' do
```

Comment:
Same here, which logic are you testing? `if submit_actions.count > 1`? I'm sure that one `if` breaks we have other problems than not rendered views ;)

Reply:
See above, same logic I'm following here

Reply:
This is a canary test for the refactoring.

Reply:
The same here, I've removed those test cases from the controller specs